### PR TITLE
CLI tools for persisting entity and POI data for dispute service. 

### PR DESCRIPTION
**Pull request by goaaron**
Refactor of #2702  to remove raw sql in place of store libraries. 

Feature additions:
- Subgraph store exposes a `find_all_versions` method that ignores block range clause
- EntityFilter can take a block number  `ChangedAtBlock`
- Layout can `deserialize_with_layout_and_block_range` which will return the block range value with the entity on query. 
- Running CLI is idempotent with new temp directories being generated

Concerns:
- Using the store to query all POI effectuates a `SELECT *`. Not clear how this will break with loading into memory.  Need a cursor or some kind of streaming ingest.

### CODE REVIEW

File: node/src/bin/manager.rs [hunk 1]
```diff
@@ -325,6 +327,59 @@ pub enum StatsCommand {
...
+        /// You can list them with "SELECT deployment from subgraphs.subgraph_deployment;"
+        #[structopt(short = "s", long)]
+        subgraph_deployment: String,
+        /// ID of the dispute
+        #[structopt(short = "d", long)]
+        dispute_id: String,
+        /// Your indexer node ID. Wallet address you supply to the indexer-agent.
+        #[structopt(short = "i", long)]
+        indexer_id: String,
+        /// Optional name of the subgraph. Find this with "SELECT name from subgraphs.subgraph;"
```
[the remark this answers is not in the document]
Comment (goaaron) [reply node/src/bin/manager.rs:346]: The ID here is the wallet ID belonging to the indexer or an operator. Currently external to graph-node internals.

File: store/postgres/src/relational_queries.rs [hunk 2]
```diff
@@ -1181,6 +1271,45 @@ impl<'a> LoadQuery<PgConnection, EntityData> for FindQuery<'a> {
 
 impl<'a, Conn> RunQueryDsl<Conn> for FindQuery<'a> {}
 
+#[derive(Debug, Clone, Constructor)]
+pub struct FindAllVersionsQuery<'a> {
```
[the remark this answers is not in the document]
Comment (goaaron) [reply store/postgres/src/relational_queries.rs:1275]: Was misusing EntityQuery.

File: store/postgres/examples/layout.rs [hunk 3]
```diff
@@ -51,7 +51,10 @@ fn print_diesel_tables(layout: &Layout) {
             ColumnType::BigDecimal | ColumnType::BigInt => "Numeric",
             ColumnType::Bytes | ColumnType::BytesId => "Binary",
             ColumnType::Int => "Integer",
-            ColumnType::String | ColumnType::Enum(_) | ColumnType::TSVector(_) => "Text",
+            ColumnType::String
+            | ColumnType::Enum(_)
+            | ColumnType::TSVector(_)
+            | ColumnType::BlockRange => "Text",
```
Comment (lutter) [inline store/postgres/examples/layout.rs:57]: The correct diesel type for `BlockRange` is `Range<Integer>`

File: store/postgres/src/chain_store.rs [hunk 4]
```diff
@@ -1,7 +1,9 @@
 use diesel::pg::PgConnection;
+
```
Comment (lutter) [inline store/postgres/src/chain_store.rs:2]: Spurious line addtion

File: store/postgres/src/chain_store.rs [hunk 5]
```diff
@@ -1,7 +1,9 @@
...
 use diesel::prelude::*;
 use diesel::r2d2::{ConnectionManager, PooledConnection};
 use diesel::sql_types::Text;
+
```
Comment (lutter) [inline store/postgres/src/chain_store.rs:6]: Spurious line addtion

**Review by lutter — CHANGES_REQUESTED** [review]

**Comment by leoyvens** [comment]
The relevance of this PR is unclear this point, so closing.

**Pull request closed without merging**